### ISSUE TITLE
Changing handle causes API tokens to be revoked

### ISSUE THREAD

**Issue by GreenFootballs** [bluesky-social/atproto#1076]
I just discovered that changing your handle will revoke an API access token. Is that supposed to happen, or is it a bug? Is the access token linked to a user's handle?

**Comment by ericsampson** [comment]
The access token has it's `sub` field set to the DID, so naively I wouldn't expect changing a handle to require revoking the token.

**Comment by GreenFootballs** [comment]
Yes, that's how it seemed to me as well, which is why I asked if it's a bug. I changed my handle to my personal domain, and the next time I tried to use the API I got an 'ExpiredToken' error with the message "Token has been revoked", and I had to create a new one.

**Comment by dholms** [comment]
That's very surprising as the handle isn't used in the auth system at all 🤔 

How did you make the refresh request? Using the `@atproto/api` library? If so it may be a client bug?

**Comment by GreenFootballs** [comment]
I changed my handle in Bluesky's settings. Then the next API call produced the "Token has been revoked" message, and I had to use the `com.atproto.server.createSession` endpoint to get a new token.

**Comment by GreenFootballs** [comment]
Oh, and no - I'm not using that library. I'm building a library myself that uses cURL without external dependencies.

**Comment by dholms** [comment]
Just out of curiosity, are you able to replicate this?

Not seeing why this would be happening 🤔 

I can try replicating as well

**Comment by GreenFootballs** [comment]
Will try and let you know.

**Comment by GreenFootballs** [comment]
OK, I tried changing my handle to a default bsky handle, then back to my custom domain handle, and it did not revoke the token this time.

However, when the token was revoked, I was trying to set up the custom domain for the first time and DNS wasn't set correctly, and it failed with error messages a couple of times. That's an additional factor that may have caused the token to get dropped.
